python/src/equistore/operations/lstsq.py

```diff
@@ -9,25 +9,104 @@
+    must have the same keys. The returned :py:class:`TensorMap` ``w`` has the
+    same keys as ``X`` and ``Y``, and stores in each block the least-squares
+    solutions :math:`w_b`.
+
+    If a block has multiple components, they are moved to the "samples" axis
+    before solving the linear system.
+
+    If gradients are present, they must be present in both ``X`` and ``Y``.
+    Gradients are concatenated with the block values along the "samples" axis,
+    :math:`A_b = [X_b, {\nabla} X_b]`, :math:`B_b = [Y_b, {\nabla} Y_b]`, and
+    the linear system :math:`A_b w_b = B_b` is solved for :math:`w_b` using
+    least-squares.
+
+    .. note::
+      The solutions :math:`w_b` differ from the output of numpy or torch in that they
+      are already transposed. Be aware of that if you want to manually access
+      the values of blocks of ``w`` (see also the example below).
+
+    Here is an example using this function:
+
+    >>> import numpy as np
+    >>> from equistore import Labels, TensorBlock, TensorMap
+    >>> import equistore
+    ...
+    >>> values_X = np.array([
+    ...     [1.0, 2.0],
+    ...     [3.0, 1.0],
+    ... ])
+    ...
+    >>> values_Y = np.array([
+    ...     [1.0, 0.0],
+    ...     [0.0, 1.0],
+    ... ])
+    ...
+    >>> samples = Labels(
+    ...     ["structure"],
+    ...     np.array([[0], [1]]),
+    ... )
+    ...
+    >>> components = []
+    ...
+    >>> properties = Labels(
+    ...     ["properties"],
+    ...     np.array([[0], [1]])
+    ... )
+    ...
+    >>> keys = Labels(names=["key"], values=np.array([[0]]))
+    ...
+    >>> block_X = TensorBlock(
+    ...     values_X, samples, components, properties,
+    ... )
+    ...
+    >>> block_Y = TensorBlock(
+    ...     values_Y, samples, components, properties,
+    ... )
+    ...
+    >>> X = TensorMap(keys, [block_X])
+    ...
+    >>> Y = TensorMap(keys, [block_Y])
+    ...
+    >>> w = equistore.lstsq(X, Y, rcond=1e-10)
+    ...
+    >>> # Note: we take the transpose here
+    >>> y = X.block(0).values @ w.block(0).values.T
+    ...
+    >>> # Set small entries in y to 0, they are numerical noise
+    >>> mask = np.abs(y) < 1e-15
+    >>> y[mask] = 0.0
+    >>> print(y)
+    [[1. 0.]
+     [0. 1.]]
+
 
     :param X: a :py:class:`TensorMap` containing the "coefficient" matrices.
-    :param Y: a :py:class:`TensorMap` containing the "dependent variable" values.
-    :param rcond: Cut-off ratio for small singular values of a. None chose the
-                default value for numpy or PyTorch
+    :param Y: a :py:class:`TensorMap` containing the "dependent variable"
+        values.
+    :param rcond: Cut-off ratio for small singular values of a. The singular
+        value :math:`{\sigma}_i` is treated as zero if smaller than
+        :math:`r_{cond}{\sigma}_1`, where :math:`{\sigma}_1` is the biggest
+        singular value of :math:`X_b`. None choses the default value for numpy
```

Comment:
I think that looks nicer?
```suggestion
        singular value of :math:`X_b`. ``None`` choses the default value for numpy
```